Sources/Common/DataModel/ImageData/index.js

```diff
@@ -393,6 +393,17 @@ function vtkImageData(publicAPI, model) {
       .getScalars()
       .getComponent(offsetIndex, comp);
   };
+
+  const superInitialize = publicAPI.initialize;
+  publicAPI.initialize = () => {
+    publicAPI.setDimensions(0, 0, 0);
```

Comment:
can we simply set the extent like the method is doing instead of creating this indirection.

Reply:
yes

Reply:
Fair enough.
I find it dangerous if tomorrow `setDimensions()` has additional side-effects...

Reply:
feel free to leave it as-is. I thought there were no reason to just set the object to its initial state instead of calling a function.